external-api/api-tests/v1/number-generator.api-test.ts

```diff
@@ -85,7 +105,124 @@ describe('getNumber', () => {
     });
   });
 
+  describe('when NUMBER_GENERATOR is set to MOCK_E2E_NUMBER_GENERATOR', () => {
+    beforeEach(() => {
+      process.env.NUMBER_GENERATOR = 'abc';
+      process.env.MOCK_E2E_NUMBER_GENERATOR = 'abc';
+    });
+
+    it('should return mocked response', async () => {
+      process.env.NUMBER_GENERATOR = 'abc';
+      process.env.MOCK_E2E_NUMBER_GENERATOR = 'abc';
+
+      // Mock request and response objects
+      const request: any = {
+        body,
+      };
+      const response: any = {
+        status: jest.fn().mockReturnThis(),
+        send: jest.fn(),
+      };
+
+      // Call the get function
+      await getNumber(request, response);
+
+      expect(response.status).toHaveBeenCalledWith(HttpStatusCode.Ok);
+
+      expect(response.send).toHaveBeenCalledWith({
+        status: HttpStatusCode.Created,
+        data: [
+          {
+            id: expect.any(Number),
+            maskedId: expect.any(String),
+            type: 1,
+            createdBy: USER.DTFS,
+            createdDatetime: new Date(nowZeroSeconds()).toISOString(),
+            requestingSystem: USER.DTFS,
+          },
+        ],
+      });
+    });
+  });
+
+  describe('when MOCK_E2E_NUMBER_GENERATOR is undefined', () => {
```

Comment:
Have left a comment, but they should not be allowed if either of them are falsy.